src/Essentials/src/MediaPicker/ImageProcessor.windows.cs

```diff
@@ -0,0 +1,150 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Windows.Graphics.Imaging;
+using Windows.Storage.Streams;
+
+namespace Microsoft.Maui.Essentials;
+
+internal static partial class ImageProcessor
+{
+	public static partial async Task<Stream> RotateImageAsync(Stream inputStream, string? originalFileName)
+	{
+		try
+		{
+			// Convert Stream to IRandomAccessStream
+			var randomAccessStream = new InMemoryRandomAccessStream();
+			await inputStream.CopyToAsync(randomAccessStream.AsStreamForWrite());
+			randomAccessStream.Seek(0);
+
+			// Create a decoder from the input stream
+			var decoder = await BitmapDecoder.CreateAsync(randomAccessStream);
+			
+			// Check if rotation is needed
+			var orientation = GetImageOrientation(decoder);
+			if (orientation == BitmapRotation.None)
+			{
+				inputStream.Position = 0;
+				return inputStream;
+			}
+
+			// Create output stream
+			var outputStream = new InMemoryRandomAccessStream();
+
+			// Create encoder
+			var encoder = await BitmapEncoder.CreateAsync(BitmapEncoder.JpegEncoderId, outputStream);
```

Comment:
Always use `JpegEncoderId` even if the original was PNG, and you ignore quality settings.

```
Guid encoderId = BitmapEncoder.JpegEncoderId;
if (Path.GetExtension(originalFileName).Equals(".png", StringComparison.OrdinalIgnoreCase))
    encoderId = BitmapEncoder.PngEncoderId;
```